### ISSUE TITLE
Issue bij installatie op RHEL7/8

### ISSUE THREAD

**Issue by VincentCCL** [LanguageMachines/ticcltools#46]
Ik krijg deze error. Ik heb een paar extra libraries geinstalleerd na eerdere errors, maar ik vind niet waar dit vandaan komt. Zowel op CentOS 7 (jerom) als op Rocky Linux 8 (nero), zelfde resultaat:

[root@jerom ticcltools]# ./build-deps.sh 

------------------------------------------------------------------------

    Building latest stable release of main dependencies from source.

------------------------------------------------------------------------

Cloning into 'ticcutils'...

remote: Enumerating objects: 3891, done.

remote: Counting objects: 100% (621/621), done.

remote: Compressing objects: 100% (266/266), done.

remote: Total 3891 (delta 355), reused 593 (delta 329), pack-reused 3270

Receiving objects: 100% (3891/3891), 1.40 MiB | 0 bytes/s, done.

Resolving deltas: 100% (2682/2682), done.

HEAD is now at aab983c... NEWS: updated for upcoming release

aclocal: installing 'm4/libtool.m4' from '/usr/share/aclocal/libtool.m4'

aclocal: installing 'm4/ltoptions.m4' from '/usr/share/aclocal/ltoptions.m4'

aclocal: installing 'm4/ltsugar.m4' from '/usr/share/aclocal/ltsugar.m4'

aclocal: installing 'm4/ltversion.m4' from '/usr/share/aclocal/ltversion.m4'

aclocal: installing 'm4/lt~obsolete.m4' from '/usr/share/aclocal/lt~obsolete.m4'

aclocal: installing 'm4/pkg.m4' from '/usr/share/aclocal/pkg.m4'

libtoolize: putting auxiliary files in `.'.

libtoolize: linking file `./ltmain.sh'

[configure.ac:41](http://configure.ac:41/): error: possibly undefined macro: AC_MSG_ERROR

      If this token and others are legitimate, please use m4_pattern_allow.

      See the Autoconf documentation.

autoreconf: /usr/bin/autoconf failed with exit status: 1

**Comment by kosloot** [comment]
Hallo @VincentCCL heb je ook het ``autoconf-archive`` geïnstalleerd?

**Comment by VincentCCL** [comment]
Bedankt, met het autoconf-archive pakket erbij begint hij wel te compileren. Op CentOS 7 stopt hij met deze error:

/bin/sh ../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I..  -I../include -I/usr/include -I/usr/include -g -std=c++11 -O3 -W -Wall -pedantic -g -O2 -fopenmp -pthread -I/usr/include/libxml2     -MT Unicode.lo -MD -MP -MF .deps/Unicode.Tpo -c -o Unicode.lo Unicode.cxx

libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I../include -I/usr/include -I/usr/include -g -std=c++11 -O3 -W -Wall -pedantic -g -O2 -fopenmp -pthread -I/usr/include/libxml2 -MT Unicode.lo -MD -MP -MF .deps/Unicode.Tpo -c Unicode.cxx  -fPIC -DPIC -o .libs/Unicode.o

In file included from Unicode.cxx:28:0:

../include/ticcutils/Unicode.h: In function 'T TiCC::stringTo(const icu_50::UnicodeString&)':

../include/ticcutils/Unicode.h:149:14: error: 'runtime_error' is not a member of 'std'

       throw( std::runtime_error( "conversion from string '" + tmp + "' to type:"

              ^

../include/ticcutils/Unicode.h:150:15: error: must #include <typeinfo> before using typeid

      + typeid(result).name() + " failed" ) );

               ^

../include/ticcutils/Unicode.h: In function 'icu_50::UnicodeString TiCC::toUnicodeString(const T&, bool)':

../include/ticcutils/Unicode.h:170:14: error: 'runtime_error' is not a member of 'std'

       throw( std::runtime_error( std::string("conversion from type:")

              ^

../include/ticcutils/Unicode.h:171:15: error: must #include <typeinfo> before using typeid

      + typeid(obj).name()

               ^

make[2]: *** [Unicode.lo] Error 1

make[2]: Leaving directory `/tmp/build-deps.q7HhpZ/ticcutils/src'

make[1]: *** [all-recursive] Error 1

make[1]: Leaving directory `/tmp/build-deps.q7HhpZ/ticcutils'

make: *** [all] Error 2

Op Rocky Linux 8 vind ik het libtar-devel pakket niet, daarop faalt hij telkens voor het compileren begint.

**Comment by proycon** [comment]
> Op Rocky Linux 8 vind ik het libtar-devel pakket niet, daarop faalt hij telkens voor het compileren begint.

Klopt, libtar-devel mist inderdaad op RHEL8-achtige systemen. Die zal je dan zelf moeten compileren en installeren helaas (van https://github.com/tklauser/libtar bv, zo deed ik het in LaMachine ook)

De compiler errors kan ik zo niet gelijk plaatsen maar RHEL7 is wel extreem oud (was ook al niet meer ondersteund vanuit LaMachine), dus het kan goed dat libstdc++ of libicu hier te oud is.

**Comment by kosloot** [comment]
In git zit nu een kleine aanpassing die misschien de compilatieproblemen oplost.
